### PR TITLE
enable SyncCreatorTokens for Existing Contracts

### PR DESCRIPTION
**Pull request by Rohan-cp**
### Summary of Changes
You can refresh collections of created tokens from
-web: NftSelector, Editor
-mobile: NftSelector

### Demo Mobile
https://github.com/gallery-so/gallery/assets/49758803/224c32c8-3611-4116-9fac-f2acf50a7440

### Demo Web
https://github.com/gallery-so/gallery/assets/49758803/60f00225-b5a9-46b9-94f7-15596c1b3b41

https://github.com/gallery-so/gallery/assets/49758803/fc418fc8-38f7-4e80-8d28-992482c06b66

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps
Can test locally with a creator

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [x] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [x] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/web/src/components/GalleryEditor/PiecesSidebar/SidebarList/RefreshContractIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+export default function RefreshContractIcon({ contractId }: Props) {
+  const [showTooltip, setShowTooltip] = useState(false);
+
+  const [syncCreatedTokensForExistingContract, isContractRefreshing] =
+    useSyncCreatedTokensForExistingContract();
+
+  const track = useTrack();
+  const handleCreatorRefreshContract = useCallback(
+    async (contractId: string) => {
+      track('Editor Sidebar: Clicked Sync Creator Tokens For Existing Contract');
```
Comment (Robinnnnn) [inline apps/web/src/components/GalleryEditor/PiecesSidebar/SidebarList/RefreshContractIcon.tsx:23]: let's make sure to fix the tracking:

```js
track('Editor Sidebar: Clicked Sync Creator Tokens For Existing Contract', {
  id: 'Refresh Single Created Contract Button',
  context: contexts.Editor,
})
```

File: apps/web/src/components/GalleryEditor/PiecesSidebar/SidebarList/CollectionTitle.tsx [hunk 2]
```diff
@@ -65,6 +73,8 @@ export default function CollectionTitle({
             selectedView={selectedView}
             setSpamPreferenceForCollection={setSpamPreferenceForCollection}
           />
+        ) : shouldDisplayContractRefreshIcon ? (
+          <RefreshContractIcon contractId={row.contractId} />
         ) : (
           <CollectionCount>{row.count}</CollectionCount>
         )}
```
Comment (Robinnnnn) [inline apps/web/src/components/GalleryEditor/PiecesSidebar/SidebarList/CollectionTitle.tsx:80]: nit: i think there's a lot of logic going on here, could simplify to:
```js
  const rightContent = useMemo(() => {
    if (selectedView === 'Created' && isMouseHovering) {
      return <RefreshContractIcon contractId={row.contractId} />;
    }
    if (selectedView !== 'Created' && isMouseHovering) {
      return (
        <ToggleSpamIcon
          row={row}
          selectedView={selectedView}
          setSpamPreferenceForCollection={setSpamPreferenceForCollection}
        />
      );
    }
    return <CollectionCount>{row.count}</CollectionCount>;
  }, [isMouseHovering, row, selectedView, setSpamPreferenceForCollection]);

...


// render
{rightContent}

```

File: apps/mobile/src/screens/NftSelectorScreen/NftSelectorContractScreen.tsx [hunk 3]
```diff
@@ -120,3 +153,68 @@ export function NftSelectorContractScreen() {
...
   );
 }
+
+type AnimatedRefreshContractIconProps = {
+  contractId: string;
+  onRefresh: () => void;
+};
+
+function AnimatedRefreshContractIcon({ contractId, onRefresh }: AnimatedRefreshContractIconProps) {
+  const { isSyncing, syncCreatedTokensForExistingContract } = useSyncTokenstActions();
```
Comment (Robinnnnn) [inline apps/mobile/src/screens/NftSelectorScreen/NftSelectorContractScreen.tsx:163]: typo

File: apps/mobile/src/screens/NftSelectorScreen/NftSelectorContractScreen.tsx [hunk 4]
```diff
@@ -120,3 +153,68 @@ export function NftSelectorContractScreen() {
...
+      size="sm"
+      onPress={handleSyncCreatedTokensForExistingContracts}
+      icon={
+        <Animated.View style={{ transform: [{ rotate: spinAnimation }] }}>
+          <RefreshIcon />
+        </Animated.View>
+      }
+      eventElementId="NftSelectorSyncCreatedTokensForContractButton"
+      eventName="Nft Selector SyncCreatedTokensForContractButton pressed"
+      eventContext={contexts.UserCollection}
```
Comment (Robinnnnn) [inline apps/mobile/src/screens/NftSelectorScreen/NftSelectorContractScreen.tsx:217]: `contexts.Posts`

File: apps/web/src/components/NftSelector/NftSelector.tsx [hunk 5]
```diff
@@ -234,6 +236,19 @@ function NftSelectorInner({ onSelectToken, headerText, preSelectedContract, even
...
 
+  const [syncCreatedTokensForExistingContract, isContractRefreshing] =
+    useSyncCreatedTokensForExistingContract();
+  const contractRefreshDisabled = filterType !== 'Created' || isContractRefreshing;
+
+  const handleCreatorRefreshContract = useCallback(async () => {
+    if (!collectionContractId) {
+      return;
+    }
+    track('NFT Selector: Clicked Sync Creator Tokens For Existing Contract');
```
Comment (Robinnnnn) [inline apps/web/src/components/NftSelector/NftSelector.tsx:247]: same thing, let's make sure to track additional details: `{ id: Refresh Single Contract Button, context: contexts.Posts }`

**Review by Robinnnnn — COMMENTED** [review]
going over this in loom but just to reiterate:
- let's make sure the mobile refresh icon is more accurate in its reaction
- let's share a generic `RefreshIcon` component
- some tracking updates
- deprecate `setCollectionContractId` and use existing state vehicles

**Review by Robinnnnn — APPROVED** [review]
